### DISCUSSION TITLE
Is possible to set  certain vertex and face unchanged when simplify glb model?

### DISCUSSION THREAD

**Discussion by Yes-Jumby** [zeux/meshoptimizer#296] [Q&A] [answered]
I have a 3D model,It is just a matrix(N*N),and I get all the vertex to triangle ，but i want to let certain vertex and face unchanged,here is the graph for comprehension
**plus:I could get all the vertex' index and all the face' vertex.**
![image](https://user-images.githubusercontent.com/51182133/119959966-10766300-bfd7-11eb-86a9-a7683517afed.png)

**Comment by zeux** [comment, marked as answer]
Just a note for the future - you can also use GitHub Discussions for questions, which helps me filter out bug reports more easily.

To answer this question, there's no way to lock the border in this way right now. It's planned as a future extension to the lower level simplification algorithm, but I don't see a good way to use it in gltfpack during glb simplification because the source file doesn't really contain the information to determine which specific vertices need to be locked.

**Reply by John-Nagle** [reply]
Unreal Engine's mesh optimizer (the Simplygon version used with UE4) has ["silhouette protection"](https://docs.unrealengine.com/4.26/en-US/WorkingWithContent/Types/StaticMeshes/LOD/), which does what the original poster seems to want. This is an important feature for quadric mesh optimization if used on thin items such as clothing. Trying to minimize the volume error tends to lose triangles at the outer edges of cloth, because they don't affect the volume error much. This makes for badly matched LODs. So you need to preserve the outer vertices somehow.

Here's an algorithm for extracting the silhouette: http://www.cs.harvard.edu/~sjg/papers/sscs.pdf
Simplygon may have a different approach.It might be enough just to identify the outermost vertices as seen in 2D from the six principal axes. Or from the 12 axes of a dodecahedron for greater precision.
